src/scraper.rs

```diff
@@ -99,7 +99,7 @@ impl Scraper {
                 let next_full_url = url.join(&next_url).unwrap();
                 let path = url_helper::to_path(&next_full_url);
 
-                if scraper.map_url_path(&next_full_url, path) && depth < scraper.args.depth {
+                if scraper.map_url_path(&next_full_url, path) && (scraper.args.depth == -1 || depth < scraper.args.depth) {
```

Comment:
```suggestion
                if scraper.map_url_path(&next_full_url, path) && (scraper.args.depth == INFINITE_DEPTH || depth < scraper.args.depth) {
```

I think it's better to add a static INFINITE_DEPTH evaluating to -1 instead of a magic value here :) But otherwise very good, thanks a lot

Reply:
You are right, that should be better now